internal/server/middleware.go

```diff
@@ -56,6 +56,37 @@ func DatabaseMiddleware(db *gorm.DB) gin.HandlerFunc {
 	}
 }
 
+func DestinationMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		destinationID := c.GetHeader("Infra-Destination")
+		if destinationID != "" {
+			if db, ok := c.MustGet("db").(*gorm.DB); ok {
+				destination, err := data.GetDestination(db, data.ByOptionalUniqueID(destinationID))
```

Comment:
I think we probably want to add a call to `access.RequireInfraRole` to make sure this has the connector role before updating the `LastSeenAt`.